src/clients/FlexbaseClient.Company.ts

```diff
@@ -0,0 +1,25 @@
+import { FlexbaseClientBase } from './FlexbaseClient.Base';
+
+interface CompanyBalance {
+    success: boolean;
+    totalInvoices: number;
+    totalPayments: number;
+    currentBalance: number;
+    creditLimit: number;
+    availableLimit: number;
+    minimumDue: number;
+    billDate: string;
+    graceDate: string;
+  }
+
+export class FlexbaseClientCompany extends FlexbaseClientBase {
+
+    async getCompanyBalance(companyId: string): Promise<CompanyBalance | null> {
+        try {
+            return await this.client.url(`/servicing/minimumDue/${companyId}`).get().json();
```

Comment:
hm, seen this in a few places - i think the function is marked async, so you may not need to necessarily `await` the call.  async marks the function and returning a promise, so the caller will be awaiting your instantiated method

Reply:
here's a bit of info on some performance impacts of doing await within an async function: https://eslint.org/docs/latest/rules/no-return-await

I'm not entirely sure, but i also might check the .json() call you are making - if this is the axios client, you're essentially making this a synchronous function

Reply:
I got it